drivers/sx127x/sx127x.c

```diff
@@ -124,8 +130,23 @@ int sx127x_init(sx127x_t *dev)
         return -SX127X_ERR_NODEV;
     }
 
+    /* Check if the reset pin is defined */
+    if (dev->params.reset_pin == GPIO_UNDEF) {
+        DEBUG("[sx127x] error: No reset pin defined.\n");
+        return -SX127X_ERR_GPIOS;
+    }
+
     _init_timers(dev);
-    xtimer_usleep(1000); /* wait 1 millisecond */
+
+    /* reset pin should be left floating during POR */
+    gpio_init(dev->params.reset_pin, GPIO_IN);
+
+    /* wait till device signals end of POR cycle */
+    while ((gpio_read(dev->params.reset_pin) > 0) ==
```

Comment:
why `> 0`? For sx1276 `SX127X_POR_ACTIVE_LOGIC_LEVEL` is `0` so shouldn't this simply be:

```
while (gpio_read(dev->params.reset_pin) == SX127X_POR_ACTIVE_LOGIC_LEVEL) {}
```

Reply:
For the sx1276 you are right. But for sx1272 it's inverted and you need to consider that the gpio api either returns 0 or some number >0

Reply:
ah, I get it - sorry for the noise